osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -311,29 +317,25 @@ private void load(OsuColour colours, LocalisationManager localisation, BeatmapDi
                 titleBinding.BindValueChanged(_ => setMetadata(metadata.Source));
                 artistBinding.BindValueChanged(_ => setMetadata(metadata.Source), true);
 
+                addInfoLabels();
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
                 starRatingDisplay.DisplayedStars.BindValueChanged(s =>
                 {
                     difficultyColourBar.Colour = colours.ForStarDifficulty(s.NewValue);
                 }, true);
 
-                starDifficulty = difficultyCache.GetBindableDifficulty(beatmapInfo, (cancellationSource = new CancellationTokenSource()).Token);
+                starDifficulty = difficultyCache.GetBindableDifficulty(working.BeatmapInfo, (cancellationSource = new CancellationTokenSource()).Token);
                 starDifficulty.BindValueChanged(s =>
                 {
                     starRatingDisplay.FadeIn(transition_duration);
                     starRatingDisplay.Current.Value = s.NewValue ?? default;
                 });
```

Comment:
This causes the star rating to roll up from zero, which it didn't do previously. I don't think we want this (makes it hard to make a quick reading of the number, and this is currently the most prominent place it's shown in the UI):

https://user-images.githubusercontent.com/191335/157644678-114577d9-3def-46c7-a731-ace992b34002.mp4

A change like this will fix the issue:

```suggestion
                starDifficulty.BindValueChanged(s =>
                {
                    starRatingDisplay.Current.Value = s.NewValue ?? default;

                    // Don't roll the counter on initial display (but still allow it to roll on applying mods etc.)
                    if (!starRatingDisplay.IsPresent)
                        starRatingDisplay.FinishTransforms(true);

                    starRatingDisplay.FadeIn(transition_duration);
                });
```

Reply:
Good catch.